services/galaxytoolshed/galaxy-toolshed-base.js

```diff
@@ -0,0 +1,56 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const orderedInstallableRevisionsSchema = Joi.array()
+  .items(Joi.string())
+  .required()
+
+export default class BaseGalaxyToolshedService extends BaseJsonService {
+  static defaultBadgeData = { label: 'galaxy-toolshed' }
+  static baseUrl = 'https://toolshed.g2.bx.psu.edu'
+
+  async fetchOrderedInstallableRevisionsSchema({ repository, owner }) {
+    return this._requestJson({
+      schema: orderedInstallableRevisionsSchema,
+      url: `${this.constructor.baseUrl}/api/repositories/get_ordered_installable_revisions?name=${repository}&owner=${owner}`,
+    })
+  }
+
+  async fetchRepositoryRevisionInstallInfoSchema({
+    repository,
+    owner,
+    schema,
+    changesetRevision,
+  }) {
+    return this._requestJson({
+      schema,
+      url: `${this.constructor.baseUrl}/api/repositories/get_repository_revision_install_info?name=${repository}&owner=${owner}&changeset_revision=${changesetRevision}`,
+    })
+  }
+
+  async fetchLastOrderedInstallableRevisionsSchema({
+    repository,
+    owner,
+    schema,
+  }) {
+    const changesetRevisions =
+      await this.fetchOrderedInstallableRevisionsSchema({
+        repository,
+        owner,
+      })
+    return this.fetchRepositoryRevisionInstallInfoSchema({
+      repository,
+      owner,
+      schema,
+      changesetRevision: changesetRevisions.shift(),
```

Comment:
My understanding is that we just need to get the revision id/value from the first element in the array (which I'm also assuming will always have at least one element as per above discussion on the schema).

If that's indeed the case, we should just grab the first element instead of grabbing it and modifying the entire collection, i.e.:

```suggestion
      changesetRevision: changesetRevisions[0],
```

the `shift()` function does result in the first element of the array being returned, but it does so by popping that element off the top and then rearranging all the other elements which would be unnecessary and wasteful work (even if it would only have minimal impact in practice)